### ISSUE TITLE
Docker ARM64 Image

### ISSUE THREAD

**Issue by joseluisq** [joseluisq/static-web-server#54]
**Is your feature request related to a problem? Please describe.**

No

**Describe the solution you'd like**
We would like to be able to use `static-web-server` on ARM 64 Docker environments.

**Describe alternatives you've considered**

Probably we want to revisit https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/

**Additional context**
No

**Comment by talesam** [comment]
```
standard_init_linux.go:228: exec user process caused: exec format error
standard_init_linux.go:228: exec user process caused: exec format error
standard_init_linux.go:228: exec user process caused: exec format error
standard_init_linux.go:228: exec user process caused: exec format error
standard_init_linux.go:228: exec user process caused: exec format error
standard_init_linux.go:228: exec user process caused: exec format error
```

I'm running docker-compose on an ARM64 like this:

```
version: "3.9"

services:
  sws-veidahavan:
    image: joseluisq/static-web-server:2
    container_name: sws-veidahavan
    restart: always
    #environment:
      # Nota: esses envs são personalizáveis, mas também opcionais
      #- SERVER_HOST=127.0.0.1
      #- SERVER_PORT=80
      #- SERVER_ROOT=/public
    volumes:
      - /home/tales/Docker/SWS/assets.veidahavan.com/html:/public
    #labels:
    #  - "traefik.enable=true"
    #  - "traefik.frontend.entryPoints=https"
    #  - "traefik.backend=localhost_dev"
    #  - "traefik.frontend.rule=Host:localhost.dev"
    #  - "traefik.port=80"
    expose:
      - 80
    networks:
      t4l35-net:

networks:
  t4l35-net:
      name: t4l35-net
```

I looked in the documentation and it said that the image is ARM64 compatible, can you help me?
https://sws.joseluisq.net/platforms-architectures/

**Comment by joseluisq** [comment]
> I looked in the documentation and it said that the image is ARM64 compatible, can you help me?
> https://sws.joseluisq.net/platforms-architectures/

Basically what the docs say is that sws provides `arm64` binaries as well as others architectures.
BUT what this issue says effectively that sws misses the `arm64` Docker Image one on Docker Hub. Since we are only providing `amd64` ones.

But don't worry until that. The workarounds for now are:
- Just append `platform: linux/amd64` to your service on your Docker Compose file
- Or build your own image using the `arm64` binary for Docker on ARM64.

**Comment by talesam** [comment]
Error: `wget: server returned error: HTTP/1.1 404 Not Found
The command '/bin/sh -c set -ex;        wget --quiet -O /tmp/static-web-server.tar.gz "https://github.com/joseluisq/static-web-server/releases/download/v$SERVER_VERSION/static-web-server-v$SERVER_VERSION-x86_64-unknown-linux-musl.tar.gz";     tar xzvf /tmp/static-web-server.tar.gz;     cp static-web-server-v${SERVER_VERSION}-x86_64-unknown-linux-musl/static-web-server /usr/local/bin/;         rm -rf /tmp/static-web-server.tar.gz static-web-server-v${SERVER_VERSION}-x86_64-unknown-linux-musl;  chmod +x /usr/local/bin/static-web-server' returned a non-zero code: 1`


```
tales@t4l35-orion:~/GitHub/static-web-server/docker/alpine$ docker build -t talesam/static-web-server:latest .
Sending build context to Docker daemon  4.096kB
Step 1/13 : FROM alpine:3.14
3.14: Pulling from library/alpine
be307f383ecc: Pull complete 
Digest: sha256:635f0aa53d99017b38d1a0aa5b2082f7812b03e3cdb299103fe77b5c8a07f1d2
Status: Downloaded newer image for alpine:3.14
 ---> 5b8b7b635229
Step 2/13 : ARG SERVER_VERSION=0.0.0
 ---> Running in 1f493bde8c00
Removing intermediate container 1f493bde8c00
 ---> 2dce134a8cc6
Step 3/13 : ENV SERVER_VERSION=${SERVER_VERSION}
 ---> Running in ee0efc75492e
Removing intermediate container ee0efc75492e
 ---> fb3b6104e9f0
Step 4/13 : LABEL version="${SERVER_VERSION}"     description="A blazing fast and asynchronous web server for static files-serving."     maintainer="Jose Quintana <joseluisq.net>"
 ---> Running in 65be69ac6966
Removing intermediate container 65be69ac6966
 ---> eb64077e91dc
Step 5/13 : RUN apk --no-cache add ca-certificates tzdata
 ---> Running in 6c8f00dbca37
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/aarch64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/aarch64/APKINDEX.tar.gz
(1/2) Installing ca-certificates (20211220-r0)
(2/2) Installing tzdata (2021e-r0)
Executing busybox-1.33.1-r6.trigger
Executing ca-certificates-20211220-r0.trigger
OK: 9 MiB in 16 packages
Removing intermediate container 6c8f00dbca37
 ---> 368715a91026
Step 6/13 : RUN set -ex;        wget --quiet -O /tmp/static-web-server.tar.gz "https://github.com/joseluisq/static-web-server/releases/download/v$SERVER_VERSION/static-web-server-v$SERVER_VERSION-x86_64-unknown-linux-musl.tar.gz";     tar xzvf /tmp/static-web-server.tar.gz;     cp static-web-server-v${SERVER_VERSION}-x86_64-unknown-linux-musl/static-web-server /usr/local/bin/;         rm -rf /tmp/static-web-server.tar.gz static-web-server-v${SERVER_VERSION}-x86_64-unknown-linux-musl;  chmod +x /usr/local/bin/static-web-server
 ---> Running in d14c2c2515cd
+ wget --quiet -O /tmp/static-web-server.tar.gz https://github.com/joseluisq/static-web-server/releases/download/v0.0.0/static-web-server-v0.0.0-x86_64-unknown-linux-musl.tar.gz
wget: server returned error: HTTP/1.1 404 Not Found
The command '/bin/sh -c set -ex;        wget --quiet -O /tmp/static-web-server.tar.gz "https://github.com/joseluisq/static-web-server/releases/download/v$SERVER_VERSION/static-web-server-v$SERVER_VERSION-x86_64-unknown-linux-musl.tar.gz";     tar xzvf /tmp/static-web-server.tar.gz;     cp static-web-server-v${SERVER_VERSION}-x86_64-unknown-linux-musl/static-web-server /usr/local/bin/;         rm -rf /tmp/static-web-server.tar.gz static-web-server-v${SERVER_VERSION}-x86_64-unknown-linux-musl;  chmod +x /usr/local/bin/static-web-server' returned a non-zero code: 1
```

**Comment by joseluisq** [comment]
@talesam follow the below steps to build a Linux ARM64 Docker image (Alpine ARM64).

1. Replace the following lines of the `docker/alpine/Dockerfile`

https://github.com/joseluisq/static-web-server/blob/5f9f9f9cbb4ca342bcade4ae7406d152bae53b0f/docker/alpine/Dockerfile#L11-L16

2. And use these lines below instead which are adjusted to use `aarch64-unknown-linux-musl` (ARM64)

```Dockerfile
RUN set -ex; \
	wget --quiet -O /tmp/static-web-server.tar.gz "https://github.com/joseluisq/static-web-server/releases/download/v$SERVER_VERSION/static-web-server-v$SERVER_VERSION-aarch64-unknown-linux-musl.tar.gz"; \
    tar xzvf /tmp/static-web-server.tar.gz; \
    cp static-web-server-v${SERVER_VERSION}-aarch64-unknown-linux-musl/static-web-server /usr/local/bin/; \
	rm -rf /tmp/static-web-server.tar.gz static-web-server-v${SERVER_VERSION}-x86_64-unknown-linux-musl; \
	chmod +x /usr/local/bin/static-web-server
```

3. Save the Dockerfile and finally build the image from your current `static-web-server/` directory

```sh
docker build -t joseluisq/static-web-server --build-arg SERVER_VERSION=2.4.0 -f docker/alpine/Dockerfile .
```

**Comment by talesam** [comment]
generated!

https://hub.docker.com/repository/docker/talesam/static-web-server

**Comment by joseluisq** [comment]
**FYI:** The upcoming `v2.6.0` will support multiple Docker platforms included ARM ones.
You can give it a try using the latest [v2.6.0-beta.2](https://hub.docker.com/r/joseluisq/static-web-server/tags?page=1&name=2.6.0-beta.2) on Docker Hub.

**Comment by joseluisq** [comment]
Released on [v2.6.0](https://github.com/joseluisq/static-web-server/releases/tag/v2.6.0)
